### PR TITLE
Set CI and other things necessary for the base repo

### LINKED ISSUE #18: Base repo structure and CI
- Add a basic CI that builds, run tests , runs fmt, clippy, etc.
- Add contribution guidelines
- Add security policy

### PR DESCRIPTION
**Pull request by ggonzalez94**
Resolves #18

### CODE REVIEW

**Comment by ggonzalez94** [comment]
Unfortunately all checks will fail since we don't have any `Cargo.toml` file or rust files at the moment. We will have to test it once this is merged to main, or alternatively we could open the PR against the current development branch

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          tool: cargo-nextest
+
+      - name: Run tests
+        run: cargo nextest run --release
+
+      - name: Check benchmarking compilation
+        run: cargo check --release --features runtime-benchmarks
+
+      - name: Run tests with async backing
+        run: cargo nextest run --release --features="async-backing"
```
Comment (ggonzalez94) [inline .github/workflows/test.yml:63]: Do we need these two for this repo? I know we had them on the pallet templates repo, but not sure how relevant they are here

Comment (ozgunozerk) [reply .github/workflows/test.yml:63]: yes we don't

**Comment by ggonzalez94** [comment]
Merging this so we have a base repo structure and CI and if something is failing when a PR is run against it I can come back and fix it.

**Pull request merged**